src/Adapter/Address/CommandHandler/EditCustomerAddressHandler.php

```diff
@@ -57,12 +60,32 @@ public function handle(EditCustomerAddressCommand $command): void
                 throw new AddressConstraintException('Address contains invalid field values');
             }
 
-            if (false === $address->update()) {
+            // The address is used by an order so it is not edited directly, instead a copy is created and
+            if ($address->isUsed()) {
+                // Get a copy of current address
+                $old_address = new Address($address->id);
+
+                // Reset ID to force recreating a new address
+                $address->id = $address->id_address = null;
+
+                // We consider this address as necessarily NOT deleted, in case you were editing a deleted address
+                // from an order then the newly edited address should not be deleted, so that you can select it
+                $address->deleted = 0;
```

Comment:
This means, if we edit a deleted address, it's automatically undeleted? should be `if ($address->isUsed() && !$address->deleted)` or something similar?

Reply:
Yeh I wasn't 100% sure about this one, but here is the use case: let's you have an order with invoice and delivery address with id 4. You edit the delivery address, so the address with id 4 becomes deleted, and a new address is inserted

Now you also want to edit the invoice address which is still the address with id 4 (deleted), you create a new address by editing it, but since this new address is based on a deleted address the new address is also deleted.

I don't think this is what the user expects, a newly edited and created address should not be deleted I guess. And worst case scenario the new address can always be deleted in the list anyway.

What do you think?

Reply:
Oh and by the way regarding the condition you proposed, it changes nothing as we are already in a block `if ($address->isUsed())` And i we only set `deleted = 0` when `deleted == 0` then it changes nothing
The real question is indeed should we undelete it by default? as explained in the use case above.